firebase-common/src/main/java/com/google/firebase/concurrent/FirebaseExecutors.java

```diff
@@ -33,6 +35,91 @@ public static Executor newSequentialExecutor(Executor delegate) {
     return new SequentialExecutor(delegate);
   }
 
+  /**
+   * Returns a {@link Executor} that limits the number of running tasks at a given time.
+   *
+   * <p>The executor uses delegate in order to {@link Executor#execute(Runnable) execute} each task,
+   * and does not create any threads of its own.
+   *
+   * @param delegate {@link Executor} used to execute tasks
+   * @param concurrency max number of tasks executing concurrently
+   */
+  public static Executor newLimitedConcurrencyExecutor(Executor delegate, int concurrency) {
+    return new LimitedConcurrencyExecutor(delegate, concurrency);
+  }
+
+  /**
+   * Returns a {@link ExecutorService} that limits the number of running tasks at a given time.
+   *
+   * <p>The executor uses delegate in order to {@link Executor#execute(Runnable) execute} each task,
+   * and does not create any threads of its own.
+   *
+   * @param delegate {@link ExecutorService} used to execute tasks
+   * @param concurrency max number of tasks executing concurrently
+   */
+  public static ExecutorService newLimitedConcurrencyExecutorService(
+      ExecutorService delegate, int concurrency) {
+    return new LimitedConcurrencyExecutorService(delegate, concurrency);
+  }
+
+  /**
+   * Returns a {@link ScheduledExecutorService} that limits the number of running tasks at a given
+   * time.
+   *
+   * <p>The executor uses delegate in order to {@link Executor#execute(Runnable) execute} each task,
+   * and does not create any threads of its own.
+   *
+   * @param delegate {@link ExecutorService} used to execute tasks
+   * @param concurrency max number of tasks executing concurrently
+   */
+  public static ScheduledExecutorService newLimitedConcurrencyScheduledExecutorService(
+      ExecutorService delegate, int concurrency) {
+    return new DelegatingScheduledExecutorService(
+        newLimitedConcurrencyExecutorService(delegate, concurrency),
+        ExecutorsRegistrar.SCHEDULER.get());
+  }
+
+  /**
+   * Returns a {@link PausableExecutor }.
```

Comment:
I think it makes sense to document that when paused, currently running task do not stop. wdyt?